### ISSUE TITLE
v1.6.1 causes HA to hang on restart

### ISSUE THREAD

**Issue by nsleigh** [hugobloem/stateful_scenes#170]
I think this is related to v1.6.1 - I upgraded from 1.6.0 and when I restarted HA I couldn't connect to the UI. I rolled back to 1.6.0 by replacing the file and now it is ok. I can't see anything in the logs but then I am not really sure what to look for.

**Comment by choenig** [comment]
same here, reverted to 1.6.0, now Home Assistant GUI shows up again.

**Comment by cayossarian** [comment]
I believe this may be related to the incorrect restoration of the 'off scene' select which I was just looking at.  I'll get a PR going shortly.

**Comment by cayossarian** [comment]
If anybody can give this is a spin it would help before Hugo pulls the fix: [https://github.com/cayossarian/stateful_scenes/tree/fix/off_scene_select_restore_fixes](https://github.com/cayossarian/stateful_scenes/tree/fix/off_scene_select_restore_fixes)

**Comment by SwishSwushPow** [comment]
Just a bit more input: I've not experienced any issues, but I've also never touched the "off scene" setting.

**Comment by hugobloem** [comment]
I couldn't test these features as I am away at the moment. 

> If anybody can give this is a spin it would help before Hugo pulls the fix: [https://github.com/cayossarian/stateful_scenes/tree/fix/off_scene_select_restore_fixes](https://github.com/cayossarian/stateful_scenes/tree/fix/off_scene_select_restore_fixes)
> 
> 

Did this solve things? Let me know and I'll merge.

**Comment by millerim** [comment]
HA still hangs on startup with the fix.  Let me know what you need to troubleshoot.

**Comment by nsleigh** [comment]
> If anybody can give this is a spin it would help before Hugo pulls the fix: [https://github.com/cayossarian/stateful_scenes/tree/fix/off_scene_select_restore_fixes](https://github.com/cayossarian/stateful_scenes/tree/fix/off_scene_select_restore_fixes)
> 
> 

Sorry away at the moment.

**Comment by hmdrmch** [comment]
How do you revert back to 1.6.0? I have cli access to my Home Assistant only, but i’ve never worked with the cli before.

**Comment by nsleigh** [comment]
> How do you revert back to 1.6.0? I have cli access to my Home Assistant only, but i’ve never worked with the cli before.

The way I did it was to connect to the SMB share of the drive and overwrite the files. Do you have share access?

**Comment by cayossarian** [comment]
I can’t get my instance to hang.  Any logs would help or describe how you may have configured the integration before the hang.   To downgrade you can manually copy the source 1.60 files from the release zip into the corresponding config/custom_components directory on your HA instance

**Comment by cayossarian** [comment]
Maybe try to quit the browser and connect again?

**Comment by fugazzy** [comment]
v1.6.2 does not solve the problem for me. I do not know how to provide more info, as the complete UI does not show up, when version 1.6.1 or 1.6.2 are active.

**Comment by millerim** [comment]
Same.  Happy to help, but not sure how.

**Comment by pcprinz** [comment]
Same here with 1.6.2. HA hangs up on restart at booting zha - or at least thats the last step before the gui vanishes. I had to restore a backup from the night before, where it was still 1.6.0 and now it's working again

**Comment by tahlequawesome** [comment]
Just ran into this last night. Went from 1.6.0 to 1.6.2 and HA never came up. Ended up renaming the folder via SMB share and restarting, then downloading 1.6.0 again, after which HA started as normal.
As with those above me, I would be happy to assist with information gathering if someone can point me to what would be helpful.
Thank you for this really innovative integration!

**Comment by cayossarian** [comment]
Can someone that is having a hang please uninstall your current integration and try this custon repository.  i have streamlined the fix for the race condition to avoid use of a particular thread method in favor of asynchronous evaluation.  If it works we will push a PR

[https://github.com/cayossarian/stateful_scenes/tree/fix/avoid_multiple_threads_on_same_scene](https://github.com/cayossarian/stateful_scenes/tree/fix/avoid_multiple_threads_on_same_scene)

**Comment by millerim** [comment]
I wasn’t thrilled at the prospect of uninstalling the integration and switching to a custom repository, but I downloaded your code and copied it over the contents of the stateful_scene directory and restarted HA.  It started up successfully and Stateful Scenes seems to work.

**Comment by cayossarian** [comment]
yes, I get your hesitation with yet another repository install.  If Hugo wasn't on holiday we could have worked a beta here.  Really appreciate your efforts though since I couldn't reproduce the hang and the PR is now in Hugo's queue to release.  Happy New Year!

**Comment by AndreaCCIE** [comment]
> Can someone that is having a hang please uninstall your current integration and try this custon repository. 

I replaced the folder as @millerim mentioned, but it didn't work for me. I had to remove the folder and reinstall 1.6.0.

**Comment by millerim** [comment]
I copied all the files into the state_scenes folder, copying over the previous files. On Dec 31, 2024, at 7:54 PM, Andrea ***@***.***> wrote:﻿

Can someone that is having a hang please uninstall your current integration and try this custon repository.

I replaced the folder as @millerim mentioned, but it didn't work for me. I had to remove the folder and reinstall 1.6.0.

**Comment by cayossarian** [comment]
> > Can someone that is having a hang please uninstall your current integration and try this custon repository.
> 
> I replaced the folder as @millerim mentioned, but it didn't work for me. I had to remove the folder and reinstall 1.6.0.

One can also use HAC install custom test repository (three dots upper/right) but still need to be ready to downgrade manually.  When a hang occurs look in the log if you are at the terminal, even a hang should eventually throw an exception if it's a scheduling problem.  I don't see any entries.  At the risk of repeating myself clear the browser cache and reload the page when it's hanging.

**Comment by cayossarian** [comment]
updated the branch to test.

**Comment by hugobloem** [comment]
Thanks for confirming it works and huge thanks to @cayossarian for doing the hard work in rewriting the code and fixing the bugs! 

I am releasing 1.6.3 which includes the latest changes. 

Happy new year, Hugo.

**Comment by millerim** [comment]
Updated to 1.6.3 and restarted successfully.  I'll keep an eye on it.  Thanks all and HNY!

**Comment by fugazzy** [comment]
> v1.6.2 does not solve the problem for me. I do not know how to provide more info, as the complete UI does not show up, when version 1.6.1 or 1.6.2 are active.

I can confirm that v1.6.3 now works again. Thank you and a happy new year!

**Comment by nsleigh** [comment]
Sorry I wasn't around to test this after reporting the issue. Resolved for me now, thank you to those involved.
